server/src/com/cloud/agent/manager/allocator/impl/FirstFitAllocator.java

```diff
@@ -297,6 +297,7 @@
                     s_logger.debug("Host name: " + host.getName() + ", hostId: " + host.getId() +
                         " already has max Running VMs(count includes system VMs), skipping this and trying other available hosts");
                 }
+                avoid.addHost(host.getId());
```

Comment:
@SudharmaJain There is another check for GPU device immediately below and the pattern is similar to the guest VM check. The issue you have mentioned can happen even for this. Can you check and verify?